include/numerics/dense_matrix.h

```diff
@@ -34,6 +34,21 @@
 #include <vector>
 #include <algorithm>
 
+#ifdef LIBMESH_HAVE_METAPHYSICL
+namespace MetaPhysicL
+{
+template <typename, typename>
+class DualNumber;
```

Comment:
I guess this is equivalent to
```
template <typename T, typename D> class DualNumber<T,D>;
```
I haven't seen your version (with unnamed template parameters) before...

Reply:
correct